call.go

```diff
@@ -20,13 +20,21 @@ package grpc
 
 import (
 	"context"
+
+	"google.golang.org/grpc/metadata"
+	imetadata "google.golang.org/grpc/internal/metadata"
 )
 
 // Invoke sends the RPC request on the wire and returns after response is
 // received.  This is typically called by generated code.
 //
 // All errors returned by Invoke are compatible with the status package.
 func (cc *ClientConn) Invoke(ctx context.Context, method string, args, reply interface{}, opts ...CallOption) error {
+	if md, _ ,ok := metadata.FromOutgoingContextRaw(ctx); ok {
```

Comment:
This check should be done in `newClientStream()` so it's handled for both unary and streaming RPCs